include/oneapi/dpl/internal/dynamic_selection_impl/auto_tune_policy.h

```diff
@@ -252,59 +227,40 @@ class auto_tune_policy
         }
     }
 
-    template <typename Function, typename... Args>
-    auto
-    submit(selection_type e, Function&& f, Args&&... args)
+    void
+    initialize_impl(timing_t resample_time = never_resample)
     {
-        static_assert(sizeof...(KeyArgs) == sizeof...(Args));
-        if (backend_)
-        {
-            return backend_->submit(e, std::forward<Function>(f), std::forward<Args>(args)...);
-        }
-        else
+        if (!state_)
         {
-            throw std::logic_error("submit called before initialization");
+            state_ = std::make_shared<state_t>();
+            resample_time_ = resample_time;
+            auto u = base_t::get_resources();
+            for (size_type i = 0; i < u.size(); ++i)
+            {
+                state_->resources_with_index_.push_back(resource_with_index_t{u[i], i});
+            }
         }
     }
 
-    auto
-    get_resources()
-    {
-        if (backend_)
-        {
-            return backend_->get_resources();
-        }
-        else
-        {
-            throw std::logic_error("get_resources called before initialization");
-        }
-    }
+  public:
+    // Needed by Policy Traits
+    using resource_type = decltype(unwrap(std::declval<wrapped_resource_t>()));
 
-    auto
-    get_submission_group()
+    auto_tune_policy(deferred_initialization_t) {}
```

Comment:
Should we mark this constructor as `explicit` ?

Reply:
I'm not sure what we benefit from this.  I suppose we could, but basically we are preventing 
```auto_tune_policy p = oneapi::dpl::experimental::deferred_initialization;```
Is there another advantage I'm missing here?